### PR TITLE
Add assignee removal/addition support to PRs

### LINKED ISSUE #1009: How to assign a user to a pull request recently created?
image

```python
fork_user = github_api.get_user('user')
fork_repo = fork_user.get_repo('repo')

fork_pullrequest = fork_repo.create_pull("Title", "Body", 'master', 
          '{}:{}'.format('otheruser', 'master'), False)
```

After running this code, I got the pull_request object. Looking over its documentation at:
1. https://pygithub.readthedocs.io/en/latest/github_objects/PullRequest.html

I could not find a method to assign a user to the pull request like:
```python
fork_pullrequest.assign(fork_user)
```

### LINKED ISSUE #1109: Rapid decrease of search API rate limit after 1 function call
I am using the search_code function and am occasionally seeing the remaining rate limit for the search API decrease by more than 1 after the function is called. According to the Github API documentation, it should decrease by 1 per request and are allowed 30 requests per minute. So my 1 request using search_code shouldn't be exceeding the rate limit however it often does. Has anyone else experienced this kind of issue or are there any ideas for why this function call is showing as more than 1 request against the rate limit?

### PR DESCRIPTION
**Pull request by nagapavan**
Add assignee removal/addition support to PRs
Fixes #1009 

Add test for new PR assignee functionality

Add record file

Addressed comments on #1016 and fixed test case

### CODE REVIEW

**Comment by nagapavan** [comment]
@llimllib @s-t-e-v-e-n-k @rallytime Please review this PR

**Comment by llimllib** [comment]
Am I related to this PR somehow? If so I don’t remember

**Comment by nagapavan** [comment]
@llimllib Sorry, I have added you accidentally.

File: github/PullRequest.py [hunk 1]
```diff
@@ -794,6 +794,36 @@ def merge(self, commit_message=github.GithubObject.NotSet, commit_title=github.G
         )
         return github.PullRequestMergeStatus.PullRequestMergeStatus(self._requester, headers, data, completed=True)
 
+    def add_to_assignees(self, *assignees):
+        """
+        :calls: `POST /repos/:owner/:repo/issues/:number/assignees <https://developer.github.com/v3/issues/assignees>`_
+        :param assignee: :class:`github.NamedUser.NamedUser` or string
```
Comment (s-t-e-v-e-n-k) [inline github/PullRequest.py]: This is not the parameter. You are expecting a list and even iterating over it, please correct the docstring.

File: github/PullRequest.py [hunk 2]
```diff
@@ -794,6 +794,36 @@ def merge(self, commit_message=github.GithubObject.NotSet, commit_title=github.G
...
+            "POST",
+            self.url + "/assignees",
+            input=post_parameters
+        )
+        self._useAttributes(data)
+
+    def remove_from_assignees(self, *assignees):
+        """
+        :calls: `DELETE /repos/:owner/:repo/issues/:number/assignees <https://developer.github.com/v3/issues/assignees>`_
+        :param assignees: :class:`github.NamedUser.NamedUser` or string
```
Comment (s-t-e-v-e-n-k) [inline github/PullRequest.py]: Here the name is correct, but it's still an array you're expecting. Passing a string will not behave like a user expects.

File: tests/PullRequest.py [hunk 3]
```diff
@@ -218,3 +239,14 @@ def testMerge(self):
 
     def testMergeWithCommitMessage(self):
         self.g.get_user().get_repo("PyGithub").get_pull(39).merge("Custom commit message created by PyGithub")
+
+    def testAddAndRemoveAssignees(self):
+        user1 = "jayfk"
+        user2 = self.g.get_user("jzelinskie")
+        self.assertListKeyEqual(self.pull.assignees, lambda a: a.login, ["stuglaser", "jacquev6"])
+        # pr_issue = self.pull.as_issue()
```
Comment (s-t-e-v-e-n-k) [inline tests/PullRequest.py]: Why is this commented out?

File: tests/PullRequest.py [hunk 4]
```diff
@@ -160,10 +161,30 @@ def testEditWithAllArguments(self):
...
+                                 "src/github/Issue.py", "src/github/IssueComment.py", "src/github/IssueEvent.py",
+                                 "src/github/Label.py", "src/github/Milestone.py", "src/github/NamedUser.py",
+                                 "src/github/Organization.py", "src/github/Permissions.py", "src/github/Plan.py",
+                                 "src/github/PullRequest.py", "src/github/PullRequestComment.py",
+                                 "src/github/PullRequestFile.py", "src/github/Repository.py",
+                                 "src/github/RepositoryKey.py", "src/github/Tag.py", "src/github/Team.py",
+                                 "src/github/UserKey.py", "test/Issue.py", "test/IssueEvent.py",
+                                 "test/ReplayData/Issue.testAddAndRemoveLabels.txt",
+                                 "test/ReplayData/Issue.testDeleteAndSetLabels.txt",
+                                 "test/ReplayData/Issue.testGetLabels.txt", "test/ReplayData/IssueEvent.setUp.txt"])
```
Comment (s-t-e-v-e-n-k) [inline tests/PullRequest.py]: All of these changes are beside the point of your change. Personally, I'd prefer in future if you split out things like this into a clean up commit.

Comment (nagapavan) [reply tests/PullRequest.py]: Agreed. This was added accidentally during formatting. Reverting back to original.

File: tests/PullRequest.py (hunk 3 above)
Comment (nagapavan) [reply tests/PullRequest.py]: Cleaned up the comment.

File: github/PullRequest.py (hunk 1 above)
Comment (nagapavan) [reply github/PullRequest.py]: @s-t-e-v-e-n-k corrected now. Was a copy-paste mistake, will pay more attention next time.

File: github/PullRequest.py (hunk 2 above)
Comment (nagapavan) [reply github/PullRequest.py]: @s-t-e-v-e-n-k Corrected now.

**Comment by nagapavan** [comment]
@s-t-e-v-e-n-k Thanks for the comments, have made changes as suggested. Will pay more attention next time and try to reduce the review iterations.

**Comment by s-t-e-v-e-n-k** [comment]
You're still removing a blank line from the setUp replay data, but I don't think I care. Thanks for your great work!

**Pull request merged**